packages/framework/aqueduct/src/data-objects/pureDataObject.ts

```diff
@@ -72,11 +73,20 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
      */
     public get handle(): IFluidHandle<this> { return this.innerHandle; }
 
+    /**
+     * @deprecated - Going forward the data object will be accessible as the entrypoint of the data store runtime. To
+     * access it use "(runtime as FluidObject<IProvideFluidLoadable>).IFluidLoadable?.handle?.get() as PureDataObject"
+     * instead of this method. That code is a temporary workaround with discovery of properties through FluidObject
+     * until we expose entrypoints more directly.
+     */
     public static async getDataObject(runtime: IFluidDataStoreRuntime) {
-        const obj = (runtime as any)._dataObject as PureDataObject;
-        assert(obj !== undefined, 0x0bc /* "Runtime has no DataObject!" */);
-        await obj.finishInitialization(true);
```

Comment:
odd that we always call initialize here

Reply:
Yeah, as we discussed I removed that line from this method since it became part of the `initializeEntrypoint` function shown in that link from your comment, and it should now trigger only once, the first time the handle gets accessed. Which could be here, in line 87 of the new version of this file.

Reply:
Not sure why you think it's odd to call initialize here, but users (like createOdspSearchDataStoreFactory() in Bohemia) expect to get fully initialized object, and load path creates, but does not finish initialization for DataObject (as it can't - see comment in createDataObject), so this call was needed. We need to ensure we are not breaking this flow

Reply:
the odd part in not that its called. its that it is always called, which can leaded to repeated calls of the finishInitalization

Reply:
Name can be better, but it's init-once:

```ts
    public async finishInitialization(existing: boolean): Promise<void> {
        if (this.initializeP !== undefined) {
            return this.initializeP;
        }
        this.initializeP = this.initializeInternal(existing);
        return this.initializeP;
    }
```